### ISSUE TITLE
[2.1.1]  (node:11737) [DEP0131] DeprecationWarning: The legacy HTTP parser is deprecated.

### ISSUE THREAD

**Issue by kshartman** [meteor/meteor#11388]
Version 2.1.1 issues a deprecation warning for http-parser on startup.  I do not use http or request directly or indirectly.  I believe the code responsible for this is

`'use strict';

var NodeHTTPParser = process.binding('http_parser').HTTPParser,
    version        = NodeHTTPParser.RESPONSE ? 6 : 4;`

in my bundle file 

.meteor/packages/ddp-server/.2.3.2.1jl3owp.wpsuf++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/websocket-driver/lib/websocket/

running under WSL 2 (Ubuntu 20.04) but the same problem occurs on native Ubuntu 20.04.

The stack trace is 

`=> Started proxy.
(node:11737) [DEP0131] DeprecationWarning: The legacy HTTP parser is deprecated.
    at <anonymous>
    at internalBinding (internal/bootstrap/loaders.js:136:34)
    at process.binding (internal/bootstrap/loaders.js:113:14)
    at Object.<anonymous> (/home/shartman/.meteor/packages/ddp-server/.2.3.2.1jl3owp.wpsuf++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/websocket-driver/lib/websocket/http_parser.js:3:30)    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Module.Mp._compile (/home/shartman/glo/dev/.meteor/local/build/programs/server/runtime.js:50:23)    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Module.Mp.load (/home/shartman/glo/dev/.meteor/local/build/programs/server/runtime.js:15:31)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/shartman/.meteor/packages/ddp-server/.2.3.2.1jl3owp.wpsuf++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/websocket-driver/lib/websocket/driver/client.js:6:18)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Module.Mp._compile (/home/shartman/glo/dev/.meteor/local/build/programs/server/runtime.js:50:23)    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)

I don't have a simple reproduction example but others are getting the same result.  The code cited above in the websocket driver is clearly including the deprecated parser.  I suspect if you generated a legacy blaze framework app it would do the same thing.  One example in [this thread](https://forums.meteor.com/t/node-39372-dep0131-deprecationwarning-the-legacy-http-parser-is-deprecated/55670) was a fresh install of Meteor + Vue.

**Comment by rj-david** [comment]
https://nodejs.org/en/blog/release/v12.22.0/

```
The legacy HTTP parser, selected by the --http-parser=legacy command line option, is deprecated with the pending End-of-Life of Node.js 10.x (where it is the only HTTP parser implementation provided) at the end of April 2021. It will now warn on use but otherwise continue to function and may be removed in a future Node.js 12.x release.
```

**Comment by rj-david** [comment]
https://github.com/creationix/http-parser-js#readme

```
http-parser-js should work via monkey-patching on Node v6-v11, and v13-14.

Node v12.x renamed the internal http parser, and did not expose it for monkey-patching, so to be able to monkey-patch on Node v12, you must run node --http-parser=legacy file.js to opt in to the old, monkey-patchable http_parser binding.
```

**Comment by kshartman** [comment]
> https://github.com/creationix/http-parser-js#readme
> 
> ```
> http-parser-js should work via monkey-patching on Node v6-v11, and v13-14.
> 
> Node v12.x renamed the internal http parser, and did not expose it for monkey-patching, so to be able to monkey-patch on Node v12, you must run node --http-parser=legacy file.js to opt in to the old, monkey-patchable http_parser binding.
> ```

Enlighten me please.  What is "monkey patching"

**Comment by Natas0007** [comment]
I also receive this when running a freshly created 2.2 app.

![image](https://user-images.githubusercontent.com/5757764/115318813-237a5580-a144-11eb-905f-4bdf8f94004f.png)

**Comment by filipenevola** [comment]
Fixed.

If you are running latest Meteor you can update your packages to get rid of this warning:

Just run `meteor update` or update these two packages for their latest versions:

```
socket-stream-client@0.3.2.
ddp-server@2.3.3.
```

**Comment by stevengragg** [comment]
> Fixed.
> 
> If you are running latest Meteor you can update your packages to get rid of this warning:
> 
> Just run `meteor update` or update these two packages for their latest versions:
> 
> ```
> socket-stream-client@0.3.2.
> ddp-server@2.3.3.
> ```

Hi, this works for me the warning did not appear on local dev environment. But when I try to have deploy it through free testing environment of meteor (latest) through => meteor deploy  [appname].meteorapp.com --settings [settings json file] --free
I am getting a warning 
[DEP0131] DeprecationWarning: The legacy HTTP parser is deprecated.15cwr
2021-05-03 11:23:21+08:00/app/bundle/programs/server/node_modules/fibers/future.js:31315cwr
2021-05-03 11:23:21+08:00 throw(ex);
So on ....
That leads to Deployment failed

To sum up, it works on local but when deployed, its not.
Please help If you have any idea., or i missed something and it was commented on related issue
Thank you so much

**Comment by softwarecreations** [comment]
Just mentioning incase anyone else runs into this issue still...
@filipenevola I just want to mention that I'm running the versions you mentioned, but the issue persists on older versions of firefox.
A more detailed breakdown of which versions it happens in is in my post here:
https://github.com/meteor/meteor/issues/9255#issuecomment-881684974
So it seems like you need to update your Meteor stuff AND be running Firefox >= 88 to eliminate these errors.

**Comment by kwhjvdkamp** [comment]
Working out a sample found on the TopTal environment about NgRx Angular (Angular 12.2.3) in conjunction with Firebase (as backend, version 9.0) on Visual Studio Code with default Chrome browser (version 93, no Firefox) during the initial phase of building the sample 'Generating browser application bundles (phase: setup)' the erroneous message '(node:52256) [DEP0131] DeprecationWarning: The legacy HTTP parser is deprecated!' pops up, see screen shot!
![build](https://user-images.githubusercontent.com/9660415/131720077-b0c056ac-7d83-4e05-8881-ec83aa38d90c.png)
The sample builds! The assumption is, it's a remnant of something! Who can confirm... or knows a solution to get rid of it!

**Comment by StorytellerCZ** [comment]
Most likely some old package or something that uses the HTTP parser directly. Best to check your packages. What version of Meteor are you on?

**Comment by kwhjvdkamp** [comment]
No Meteor involved! The only http-parser thing involved is "http-parser-js": "^0.5.3" (according npm e.g. it's the latest)...

**Comment by StorytellerCZ** [comment]
That it is the latest doesn't mean anything. Since it is focused on http parsing it is most likely your culprit.

**Comment by Romulo-S** [comment]
Im having the same problem when do ng serve

![image](https://user-images.githubusercontent.com/1325820/134774894-8be51cd4-9039-4baa-9d19-f13aaf2f91b4.png)

**Comment by StorytellerCZ** [comment]
This could potentially be something in Angular or if you are using http_parser directly. Hard to say from this little info.

**Comment by Romulo-S** [comment]
I´m preparing to migrate to angular 12, our app is using angular 10 and i´m suspecting that have something related to this.After i complete migration, i will check if this warning persist.
Btw: This warning appears after i update to node 16.

**Comment by StorytellerCZ** [comment]
Latest Meteor is using Node 14. If you are using Angular on its own there is nothing we can do about it here.

**Comment by filipenevola** [comment]
Hi @Romulo-S, are you using Meteor in your project?

Meteor is using Node.js 14 on latest versions so I don't think you are running Node.js 16. Are you sure?

**Comment by Romulo-S** [comment]
I´m not using Meteor im my project.
![image](https://user-images.githubusercontent.com/1325820/135070012-6b034542-bd1c-4288-9be0-43f2e29d27a4.png)

**Comment by StorytellerCZ** [comment]
@Romulo-S Out of curiosity, why are you posting about this issue in a Meteor repository?

**Comment by Romulo-S** [comment]
I´m came here because i was searching about the error. "The legacy HTTP parser is deprecated"

**Comment by StorytellerCZ** [comment]
I'm afraid that is not going to work to fix your issue, you need to put your issue to the correct repository from where the error is coming from. Since it is not related to Meteor it is not in our powers to fix this. You need to identify which package is the culprit of that error message and go to their repository and ask there for them to fix it. For general help/questions I recommend using [Stackoverflow](https://stackoverflow.com/).
